### ISSUE TITLE
`punycode/` is misflagged as requiring the `node:` prefix

### ISSUE THREAD

**Issue by nbouvrette** [sindresorhus/eslint-plugin-unicorn#1931]
```js
import { encode as punycodeEncode } from 'punycode/'
```

Is reporting: 

```
Prefer `node:punycode/` over `punycode/`.eslint[unicorn/prefer-node-protocol](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/v44.0.0/docs/rules/prefer-node-protocol.md)
```

When in fact, `node:punycode` is deprecated and `punycode/` is a npm package.

**Comment by fisker** [comment]
https://github.com/sindresorhus/is-builtin-module/issues/7

**Comment by silverwind** [comment]
I think to fix this, the rule could just ignore the import if the specifier ends in `/`, which in itself seems to be some kind of hack around specific to node's module loader (I would not recommend using that, as it's bound to cause issues in other loaders).

**Comment by silverwind** [comment]
The import from `punycode/` does not work for me in node ESM, it only works in CJS, so I suggest to only support it for CJS, if at all. Better alternatives include either `punycode/punycode.js` or the `punycode.js` npm module.

```js
> await import("punycode/")
Uncaught:
Error [ERR_UNSUPPORTED_DIR_IMPORT]:

**Comment by kachkaev** [comment]
Same [here](https://github.com/hashintel/hash/pull/1704) with `import { Buffer } from "buffer/"` being converted to `import { Buffer } from "node:buffer/"`.

**Comment by fisker** [comment]
Fixed by #2060
